File: todolist/src/types/packing.ts
/** 打包清单 */
export interface PackingList {
  id: string;
  title: string;
  destination: string;
  departure_date: string | null;
  return_date: string | null;
  notes: string;
  is_template: boolean;
  created_at: string;
  updated_at: string;
  /** 已打包数量（列表接口可能返回） */
  checked_count?: number;
  /** 总物品数量（列表接口可能返回） */
  total_count?: number;
}

/** 创建清单参数 */
export interface CreatePackingListParams {
  title: string;
  destination?: string;
  departure_date?: string | null;
  return_date?: string | null;
  notes?: string;
  is_template?: boolean;
}

/** 更新清单参数 */
export interface UpdatePackingListParams {
  title?: string;
  destination?: string;
  departure_date?: string | null;
  return_date?: string | null;
  notes?: string;
  is_template?: boolean;
}

/** 打包物品 */
export interface PackingItem {
  id: string;
  list_id: string;
  name: string;
  category: string;
  quantity: number;
  is_packed: boolean;
  sort_order: number;
  notes: string;
  created_at: string;
}

/** 添加物品参数 */
export interface CreatePackingItemParams {
  list_id: string;
  name: string;
  category?: string;
  quantity?: number;
  notes?: string;
}

/** 更新物品参数 */
export interface UpdatePackingItemParams {
  name?: string;
  category?: string;
  quantity?: number;
  notes?: string;
}

/** 拖拽排序参数 */
export interface ReorderItemsParams {
  items: { id: string; sort_order: number }[];
}

/** 清单详情 */
export interface PackingListDetail {
  list: PackingList;
  items: PackingItem[];
}

/** 预设分类 */
export const PACKING_CATEGORIES: { key: string; label: string; icon: string }[] = [
  { key: 'electronics', label: '电子设备', icon: '📱' },
  { key: 'clothing', label: '衣物', icon: '👕' },
  { key: 'toiletries', label: '洗漱护肤', icon: '🧴' },
  { key: 'medicine', label: '药品', icon: '💊' },
  { key: 'documents', label: '证件', icon: '📄' },
  { key: 'other', label: '其他', icon: '🎒' },
];

export function getCategoryLabel(key: string): string {
  return PACKING_CATEGORIES.find(c => c.key === key)?.label ?? key;
}

export function getCategoryIcon(key: string): string {
  return PACKING_CATEGORIES.find(c => c.key === key)?.icon ?? '📦';
}

File: todolist/src-tauri/src/models/packing.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PackingList {
    pub id: String,
    pub title: String,
    pub destination: String,
    pub departure_date: Option<String>,
    pub return_date: Option<String>,
    pub notes: String,
    pub is_template: bool,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Deserialize)]
pub struct CreatePackingListParams {
    pub title: String,
    #[serde(default)]
    pub destination: String,
    pub departure_date: Option<String>,
    pub return_date: Option<String>,
    #[serde(default)]
    pub notes: String,
    #[serde(default)]
    pub is_template: bool,
}

#[derive(Debug, Deserialize)]
pub struct UpdatePackingListParams {
    pub title: Option<String>,
    pub destination: Option<String>,
    pub departure_date: Option<String>,
    pub return_date: Option<String>,
    pub notes: Option<String>,
    pub is_template: Option<bool>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PackingItem {
    pub id: String,
    pub list_id: String,
    pub name: String,
    pub category: String,
    pub quantity: i32,
    pub is_packed: bool,
    pub sort_order: i32,
    pub notes: String,
    pub created_at: String,
}

#[derive(Debug, Deserialize)]
pub struct CreatePackingItemParams {
    pub list_id: String,
    pub name: String,
    #[serde(default = "default_category")]
    pub category: String,
    #[serde(default = "default_quantity")]
    pub quantity: i32,
    #[serde(default)]
    pub notes: String,
}

fn default_category() -> String {
    "other".into()
}
fn default_quantity() -> i32 {
    1
}

#[derive(Debug, Deserialize)]
pub struct UpdatePackingItemParams {
    pub name: Option<String>,
    pub category: Option<String>,
    pub quantity: Option<i32>,
    pub notes: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct ReorderItemsParams {
    pub items: Vec<ItemOrderEntry>,
}

#[derive(Debug, Deserialize)]
pub struct ItemOrderEntry {
    pub id: String,
    pub sort_order: i32,
}

/// 清单详情 = 清单信息 + 物品列表（按分类分组排序）
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PackingListDetail {
    pub list: PackingList,
    pub items: Vec<PackingItem>,
}
